pkg/build/api/validation/validation.go

```diff
@@ -151,50 +161,27 @@ func hasProxy(source *buildapi.GitBuildSource) bool {
 
 func validateSource(input *buildapi.BuildSource) fielderrors.ValidationErrorList {
```

Comment:
@bparees isn't it valid to give all three types of source to a Custom build?

When I sketched out cleaning up this validation I had made `validateSource` depend on the build strategy.